### ISSUE TITLE
Loss function for graph generator

### ISSUE THREAD

**Issue by nshervt** [dmzou/SCAT#1]
I was wondering how the loss function for graph generation, equation 4 in the paper, is written as below?
```
fcn_loss_1 = tf.nn.softmax_cross_entropy_with_logits(labels=Y_features, logits=fcn(X)[0])
fcn_loss_2 = tf.nn.softmax_cross_entropy_with_logits(labels=Y_adj, logits=fcn(X)[1])
fcn_loss_2 = tf.matrix_band_part(fcn_loss_2, 0, -1) - tf.matrix_band_part(fcn_loss_2, 0, 0)

fcn_loss = tf.reduce_mean(fcn_loss_1) + 2 * tf.reduce_mean(fcn_loss_2)
```
Specifically:

1. What are upper triangular and diagonal terms doing?
2. Why loss function for the weight generator has a factor `2`?

**Comment by dmzou** [comment]
The upper triangular and diagonal terms are getting the "off-diagonal" entries. This is done for the symmetry of the generation and we don't consider the lower-triangular part. Also, the factor 2 is there since the lower-triangular part should share the same loss. Of course, the shares of fcn_loss_1 and fcn_loss_2 do not have to be 1:1, but just an artificial setup.

**Comment by nshervt** [comment]
Thanks!
